geode-core/src/test/java/org/apache/geode/test/dunit/rules/tests/SharedCountersDistributedTest.java

```diff
@@ -0,0 +1,192 @@
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import org.apache.geode.test.dunit.Host;
+import org.apache.geode.test.dunit.VM;
+import org.apache.geode.test.dunit.rules.DistributedTestRule;
+import org.apache.geode.test.dunit.rules.SharedCountersRule;
+import org.apache.geode.test.junit.categories.DistributedTest;
+
+@Category(DistributedTest.class)
+@SuppressWarnings("serial")
+public class SharedCountersDistributedTest implements Serializable {
+
+  private static final int TWO_MINUTES_MILLIS = 2 * 60 * 1000;
+  private static final String ID1 = "ID1";
+
+  private static ExecutorService executor;
+  private static CompletableFuture<Void> combined;
+  private static List<CompletableFuture<Boolean>> futures;
+
+  @ClassRule
+  public static DistributedTestRule distributedTestRule = new DistributedTestRule();
+
+  @Rule
+  public SharedCountersRule sharedCountersRule = SharedCountersRule.builder().build();
+
+  @Before
+  public void setUp() throws Exception {
+
+
+  }
+
+  @After
+  public void tearDown() throws Exception {
+
+  }
+
+  @Test
+  public void inc_withoutInit_throwsNullPointerException() throws Exception {
+    assertThatThrownBy(() -> sharedCountersRule.increment(ID1))
+        .isInstanceOf(NullPointerException.class);
+  }
+
+  @Test
+  public void ref_withoutInit_returnsNull() throws Exception {
+    assertThat(sharedCountersRule.reference(ID1)).isNull();
+  }
+
+  @Test
+  public void init_get_returnsZero() throws Exception {
+    assertThat(sharedCountersRule.initialize(ID1).reference(ID1).get()).isEqualTo(0);
+  }
+
+  @Test
+  public void initTwice_noop() throws Exception {
+    assertThat(sharedCountersRule.initialize(ID1).reference(ID1).get()).isEqualTo(0);
+  }
+
+  @Test
+  public void inc_get_returnsOne() throws Exception {
+    assertThat(sharedCountersRule.initialize(ID1).increment(ID1).reference(ID1).get()).isEqualTo(1);
+  }
+
+  @Test
+  public void incDeltaTwo_get_returnsTwo() throws Exception {
+    assertThat(sharedCountersRule.initialize(ID1).increment(ID1, 2).reference(ID1).get())
+        .isEqualTo(2);
+  }
+
+  @Test
+  public void incTwice_returnsTwo() throws Exception {
+    assertThat(
+        sharedCountersRule.initialize(ID1).increment(ID1).increment(ID1).reference(ID1).get())
+            .isEqualTo(2);
+  }
+
+  @Test
+  public void inc_getTotal_returnsOne() throws Exception {
+    sharedCountersRule.initialize(ID1).increment(ID1);
+    int total = sharedCountersRule.getTotal(ID1);
+    assertThat(total).isEqualTo(1);
+  }
+
+  @Test
+  public void inc_fromDUnitVMs_getTotal_returnsFour() throws Exception {
+    sharedCountersRule.initialize(ID1);
+    for (VM vm : Host.getHost(0).getAllVMs()) {
+      vm.invoke(() -> {
+        sharedCountersRule.increment(ID1);
+      });
+    }
+    assertThat(sharedCountersRule.getTotal(ID1)).isEqualTo(4);
```

Comment:
getAllVMs.size() instead of "4"? This goes for all the "magic" numbers here, based on the number of DUnit VMs.

Reply:
Changed to getVMCount().